Modules/Loadable/Segmentations/Logic/vtkSegmentEditorLogic.h

```diff
@@ -0,0 +1,422 @@
+/*==============================================================================
+
+Program: 3D Slicer
+
+Copyright (c) Laboratory for Percutaneous Surgery (PerkLab)
+  Queen's University, Kingston, ON, Canada. All Rights Reserved.
+
+  See COPYRIGHT.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  This file was originally developed by Csaba Pinter, PerkLab, Queen's University
+  and was supported through the Applied Cancer Research Unit program of Cancer Care
+  Ontario with funds provided by the Ontario Ministry of Health and Long-Term Care
+
+==============================================================================*/
+
+#ifndef __vtkSegmentEditorLogic_h
+#define __vtkSegmentEditorLogic_h
+
+#include "vtkSlicerSegmentationsModuleLogicExport.h"
+
+#include <string>
+#include <vector>
+#include <vtkObject.h>
+#include <vtkWeakPointer.h>
+#include <vtkSmartPointer.h>
+#include <vtkCommand.h>
+
+class vtkMRMLAbstractViewNode;
+class vtkMRMLNode;
+class vtkMRMLScalarVolumeNode;
+class vtkMRMLScene;
+class vtkMRMLSegmentEditorNode;
+class vtkMRMLSegmentationNode;
+class vtkMRMLSliceNode;
+class vtkMRMLTransformNode;
+class vtkMRMLVolumeNode;
+class vtkMatrix4x4;
+class vtkOrientedImageData;
+class vtkPolyData;
+class vtkSegment;
+class vtkSegmentation;
+class vtkSegmentationHistory;
+class vtkMRMLApplicationLogic;
+class vtkMRMLSliceLogic;
+
+/// \brief Helper segment editor logic for qMRMLSegmentEditorWidget and its associated effects
+///
+/// Provides common logic to access and modify the segmentation.
+class VTK_SLICER_SEGMENTATIONS_LOGIC_EXPORT vtkSegmentEditorLogic : public vtkObject
+{
+public:
+  static vtkSegmentEditorLogic* New();
+  vtkTypeMacro(vtkSegmentEditorLogic, vtkObject);
+
+  enum ModificationMode
+  {
+    ModificationModeSet,
+    ModificationModeAdd,
+    ModificationModeRemove,
+    ModificationModeRemoveAll
+  };
+
+  enum Events
+  {
+    SegmentationHistoryChangedEvent = vtkCommand::UserEvent + 1, //
```

Comment:
Missing comments?

Reply:
I thought it was necessary for clang-format but I was wrong 😅 
I've removed the comments!